spec/models/category_rspec.rb

```diff
@@ -3,4 +3,36 @@
 describe Category do
   it {should have_many(:videos)}
   it {should validate_uniqueness_of(:name)}
+
+  describe "#recent_videos" do
+    it "should return an empty array if no videos found" do
+      category = Category.create(name: "syfy")
+      expect(category.recent_videos).to eq([])
+    end
+    it "should return an array of one if only found one video" do
+      category = Category.create(name: "syfy")
+      stargate = Video.create(title: "stargate", description: "stargate video", category: Category.find_by(name: "syfy"))
+      expect(category.recent_videos).to eq([stargate])
+    end
+    it "should return 6 videos at most" do
+      category = Category.create(name: "syfy")
+      stargate_seasons = 10
+      stargate_seasons.times do |season|
+        stargate = Video.create(title: "stargate season #{season + 1}", description: "video from season #{season + 1}", 
+          category: Category.find_by(name: "syfy"))

```

Comment:
This part is not efficient - `category: Category.find_by(name: "syfy")` you already have that category under `category` variable, so just use it - `category: category)`